omega/omega-connector/omega-connector-grpc/src/main/java/org/apache/servicecomb/saga/omega/connector/grpc/LoadBalancedClusterMessageSender.java

```diff
@@ -150,13 +155,15 @@ private MessageSender fastestSender() {
         .filter(entry -> entry.getValue() < Long.MAX_VALUE)
         .min(Comparator.comparingLong(Entry::getValue))
         .map(Entry::getKey)
-        .orElse((event -> {
+        .orElse(event -> {
+          clusterStatus.setHealthy(false);
```

Comment:
The clusterStatus check and setting is not atomic.  maybe we should consider to use single thread to send the event message.